File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.js
function solution(s) {
    // 1. 소문자로 변환
    const lowerS = s.toLowerCase();
    
    // 2. 'p'의 개수 세기
    const pCount = [...lowerS].filter(char => char === 'p').length;
    
    // 3. 'y'의 개수 세기
    const yCount = [...lowerS].filter(char => char === 'y').length;
    
    // 4. 비교 결과 리턴 (같으면 true, 다르면 false)
    return pCount === yCount;
}

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;

        String LowerS = s.toLowerCase();
        String[] splitedLowerS = LowerS.split("");
        
        Map<String, Integer> map = new HashMap<>();
        
        for (String ch : splitedLowerS){
            map.put(ch, map.getOrDefault(ch, 0)+1);
        }
        if(map.get("p")==map.get("y")){
            return true;
        } else{
            return false;
        }
    
    }
}